subprojects/snapshots/src/main/java/org/gradle/internal/vfs/impl/SnapshotCollectingDiffListener.java

```diff
@@ -42,7 +43,7 @@ public void publishSnapshotDiff(SnapshotHierarchy.SnapshotDiffListener snapshotD
 
     private void extractRootSnapshots(FileSystemNode rootNode, Consumer<CompleteFileSystemLocationSnapshot> consumer) {
         rootNode.accept(snapshot -> {
-            if (watchFilter.test(snapshot.getAbsolutePath())) {
+            if (snapshot.getAccessType() == AccessType.DIRECT && watchFilter.test(snapshot.getAbsolutePath())) {
```

Comment:
In general I think it would be nice to statically import these enum values everywhere we use them.

Reply:
I consistently used `AccessType` in front of the enums to specify the scope. `VIA_SYMLINK` is probably unique, `DIRECT` seemed to need some specification.

Do you feel strongly about doing a static import of the constants?

Reply:
I don't feel strongly.